packages/kit/types/helper.d.ts

```diff
@@ -31,3 +31,10 @@ export type InferValue<T, Key extends keyof T, Default> = T extends Record<Key,
 	: Default;
 export type MaybePromise<T> = T | Promise<T>;
 export type Rec<T = any> = Record<string, T>;
+export type RecursiveRequired<T> = {
```

Comment:
I can't understand what this is doing. It probably deserves a code comment at least

As an alternative, should we replace `Config` with `Partial<ValidatedConfig>`?

Reply:
This is a recursive implementation of [Required](https://www.typescriptlang.org/docs/handbook/utility-types.html#requiredtype), with a few modifications to handle some of our differences between user config and validated config. I'll add a code comment for that one.

Replacing with `Partial<ValidatedConfig>` as an alternative would still need a recursive implementation of `Partial` as it only sets properties one-level deep, `ValidatedConfig` also has `kit: { files: { setup: string } }` which is only used after validation and not in the user config.